src/codeu/chat/server/View.java

```
@@ -85,6 +85,85 @@ public ServerInfo getInfo() {
+              contributions.add(model.conversationById().first(conversationPayload.id).title + " (Creator)");
+            }
+          }
+       // }
+        foundOwner.UserUpdateMap.put(foundUser.id, Time.now());
+      }
+
+      if(contributions.isEmpty()) {
+        contributions.add("(No recent conversations)");
+      }
+    // finally, update the time that status update was last requested for the specified user to now
+    foundOwner.UserUpdateMap.put(foundUser.id, Time.now());
+    } else {
+        // if foundUser is not in the current user's interests, add the note to the collection
+        contributions.add("ERROR: User is not found in interest");
+        LOG.info("ERROR: User is not found in interest.");
+      }
+    }
+    return contributions;
+
+  }
+
+  @Override
+  public int conversationStatusUpdate(String title, Uuid owner) {
+
+    int newMessages = 0;
+
+    final User foundOwner = model.userById().first(owner);
+    final ConversationHeader foundConversation = model.conversationByText().first(title);
+    final ConversationPayload foundConversationPayload = model.conversationPayloadById().first(foundConversation.id);
+    final Time lastUpdate = foundOwner.ConvoUpdateMap.get(foundConversation.id);
+    
+    if(foundConversation != null) {
+      if(foundOwner.ConvoUpdateMap.containsKey(foundConversation.id)) {
+          Message currentMessage = model.messageById().first(foundConversationPayload.firstMessage);
+          while(currentMessage != null) {
+            if(lastUpdate.compareTo(currentMessage.creation) < 0) {
+              newMessages++;
+            }
+            currentMessage = model.messageById().first(currentMessage.next);
+          }
+        foundOwner.ConvoUpdateMap.put(foundConversation.id, Time.now());
+        } else {
+          newMessages = -1;
+        }
+    } else {
+        // return some negative value to specify that conversation doesn't exist
+        newMessages = -2;
```

You could also throw an exception and then handle it in the calling function.